### PR TITLE
Invert Binary Tree || Data Structures

### LINKED ISSUE #380: Documentation for inverting binary tree
## 📚 Documentation

Documentation for inverting binary tree

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

### PR DESCRIPTION
**Pull request by aaadddiii**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

fixes #380 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

issue #380

### CODE REVIEW

**Comment by sid-1207** [comment]
Is this still open?

**Comment by aaadddiii** [comment]
Yes

File: docs/data-structures/invert-binary-tree.mdx [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+#### Output
+
+```
+Preorder traversal of the inverted binary tree is
+1 3 2 5 4
+```
+
+## Complexity Analysis
+
+- Time complexity - O(`n`) , Where `n` is the number of nodes
```
Comment (HarshCasper) [inline docs/data-structures/invert-binary-tree.mdx:252]: Keep `O(n)` in backticks

File: docs/data-structures/invert-binary-tree.mdx [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+- Space complexity - O(`n`)
```
Comment (HarshCasper) [inline docs/data-structures/invert-binary-tree.mdx:253]: Keep `O(n)` in backticks

File: docs/data-structures/invert-binary-tree.mdx [hunk 3]
```diff
@@ -0,0 +1,259 @@
+---
+title: Invert Binary Tree
+slug: /mathematics/invert-binary-tree
+---
+
+## Introduction
+
+A binary tree is a data structure in which each node has at most two children. To invert a binary tree we have to interchange the `left` and `right` child of each node. It is also known as the mirror of the binary tree
+
+
```
Comment (HarshCasper) [inline docs/data-structures/invert-binary-tree.mdx:10]: Can you remove this line

File: docs/data-structures/invert-binary-tree.mdx [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+## Explanation
+
+For inverting a binary tree we recursively swap the `left` and `right` subtrees
+Let us have an example, consider this binary tree shown below
+![image](https://i.postimg.cc/WzTd9Rp8/tree-1.png)   
+The below image of the binary tree is the binary tree obtained after inverting the binary tree in the above image
+![image](https://i.postimg.cc/prCp40pH/tree-2.png)
```
Comment (HarshCasper) [inline docs/data-structures/invert-binary-tree.mdx:17]: Use proper paragraphs right here

File: docs/data-structures/invert-binary-tree.mdx [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+## Algorithm
+
+- Swap the `right` and `left` subtrees of the binary tree
+- Recursively call the function using the left subtree until the root becomes `null`
+- Recursively call the function using the right subtree until the root becomes `null`
+
+
```
Comment (HarshCasper) [inline docs/data-structures/invert-binary-tree.mdx:25]: Remove the extra line

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by aaadddiii** [comment]
@HarshCasper  Can you please review and merge my pr

**Comment by aaadddiii** [comment]
@iamrajiv  can you please review my pr

**Review by iamrajiv — APPROVED** [review]

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**